tests/test_main.py

```diff
@@ -59,6 +59,18 @@ def test_module_invocation_latest_version() -> None:
     assert version.parse(match.group(1)) >= version.parse(__pyright_version__)
 
 
+def test_module_invocation_pylance_version() -> None:
+    proc = subprocess.run(
+        [sys.executable, '-m', 'pyright', '--version'],
+        check=True,
+        stdout=subprocess.PIPE,
+        env=dict(os.environ, PYRIGHT_PYTHON_PYLANCE_VERSION='2023.11.11'),
+    )
+    assert proc.returncode == 0
+    match = assert_matches(VERSION_REGEX, proc.stdout.decode('utf-8'))
+    assert match.group(1) == '1.1.334'
+
```

Comment:
I think it would be helpful to test the case where `latest-release` or `latest-prerelease` is given as well.

Reply:
Any suggestions on how to do that? Here are some options that occurred to me:

### Mock the file download
Somehow mock the download of the json file containing the pyright version info. This sounds great, but it's not clear to me how I would mock something in the subprocess. I didn't see any examples of this.

### Calculate the expected pyright version dynamically
Have the test download the [version metadata json file](https://github.com/microsoft/pylance-release/blob/main/releases/latest-prerelease.json) and extract the expected pyright version. Then verify that the product code output contains the same version. This would mean duplicating product code in the tests and then verifying that the product's copy does the same thing, so that seems weird and not super useful.

### Make the test verification more lenient
Rather than checking that the output contains a particular version number (the pyright version that matches the latest prerelease build of pylance), do something more lenient. Maybe just that the output contains a `VERSION_REGEX` match? So we'd be checking that the product didn't crash, but not that the output is correct. Is that useful?

Reply:
Option 3 sounds good to me, I think the value in testing `latest-release` and `latest-prerelease` is just to make sure it doesn't crash and the JSON definitely has those keys, whether or not the actual value is correct is less valuable imo and as you mentioned, much harder to test.